python/lsst/ts/salobj/sal_info.py

```diff
@@ -792,8 +797,9 @@ def _blocking_read(
             data_list = [
                 self._sample_to_data(sd, si) for sd, si in data_list if si.valid_data
             ]
-            return (reader, data_list)
-        return (None, [])
+            ret.append((reader, data_list))
```

Comment:
Please only append if data_list is not empty, e.g.:
```
if data_list:
    ret.append((reader, data_list))
```
(I probably should have done that with the old code, though it fits more neatly into the new code).

Reply:
Please make this change (or if you don't want to, please explain why -- you marked this resolved without changing it).

Reply:
Fixed now, sorry for missing it - I though that was resolved with my previous updates, which wasn't the case